src/main/scala/mesosphere/marathon/core/launcher/impl/TaskOpFactoryHelper.scala

```diff
@@ -12,20 +12,30 @@ class TaskOpFactoryHelper(
 
   private[this] val offerOperationFactory = new OfferOperationFactory(principalOpt, roleOpt)
 
-  def launch(
+  def launchEphemeral(
     taskInfo: Mesos.TaskInfo,
-    newTask: Task,
-    oldTask: Option[Task] = None): TaskOp.Launch = {
+    newTask: Task.LaunchedEphemeral): TaskOp.Launch = {
 
     assume(newTask.taskId.mesosTaskId == taskInfo.getTaskId, "marathon task id and mesos task id must be equal")
 
     def createOperations = Seq(offerOperationFactory.launch(taskInfo))
 
-    TaskOp.Launch(taskInfo, newTask, oldTask, createOperations)
+    val stateOp = TaskStateOp.Create(newTask)
+    TaskOp.Launch(taskInfo, stateOp, None, createOperations)
```

Comment:
please use a named argument for `None`, otherwise it's hard to read without looking up the argument list.